linq/UnitTest1.cs

```diff
@@ -0,0 +1,157 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace linq
+{
+    public class UnitTest1
+    {
+        public enum Suit
+        {
+            Clubs,
+            Diamonds,
+            Hearts,
+            Spades,
+        }
+    
+        public enum Rank
+        {
+            Two,
+            Three,
+            Four,
+            Five,
+            Six,
+            Seven,
+            Eight,
+            Nine,
+            Ten,
+            Jack,
+            Queen,
+            King,
+            Ace,
+        }
+    
+        public class PlayingCard
+        {
+            public Suit CardSuit { get; }
+            public Rank CardRank { get; }
+    
+            public PlayingCard(Suit s, Rank r)
+            {
+                CardSuit = s;
+                CardRank = r;
+            }
+    
+            public override string ToString()
+            {
+                return $"{CardRank} of {CardSuit}";
+            }
+        }
+    
+        public class Program
+        {
+            [Fact]
+            public static void v()
+            {
+                var startingDeck = (from s in Suits()
+                                    from r in Ranks()
+                                    select new PlayingCard(s, r))
+                                   .ToArray();
+                // check length
+                Assert.Equal(52, startingDeck.Length);
+                // assert that all of kinds are different
+                for (int i = 0;i < startingDeck.Length;i++)
+                    for (int j = i + 1;j < startingDeck.Length;j++)
+                        Assert.NotEqual(startingDeck[i], startingDeck[j]);
+    
+                var top = startingDeck.Take(26);
+                var bottom = startingDeck.Skip(26);
+    
+                var shuffle2 = top.InterleaveSequenceWith(bottom);
+                foreach (var c in shuffle2)
+                {
+                    Console.WriteLine(c);
+                }
+    
+    
+                var times = 0;
+                var shuffle = startingDeck;
+                do
+                {
+                    shuffle = shuffle.Skip(26).InterleaveSequenceWith(shuffle.Take(26)).ToArray();

```

Comment:
You can actually write an assert here as well, we know what should the resulting sequence look like. It would begin with the 27th element, then 1st, then 28th, then 2nd, then 29th, then 3rd, etc etc...
I'd make it a for loop to check whether the `shuffle[i]` is equal to...
```
if( i % 2 == 0 ) assert( secondhalf[(int)(i/2)] )
else assert( firsthalf[Math.Ceil(i/2f)] )
```
(second half being the `shuffle.Skip(26)` and the first half is `shuffle.Take(26)` - you need to create temporary arrays similarly as it is done with shuffle - `toarray`)